gradio/components/video.py

```diff
@@ -375,3 +383,100 @@ def example_payload(self) -> Any:
 
     def example_value(self) -> Any:
         return "https://github.com/gradio-app/gradio/raw/main/demo/video_component/files/world.mp4"
+
+    @staticmethod
+    def get_video_duration_ffprobe(filename: str):
+        result = subprocess.run(
+            [
+                "ffprobe",
+                "-v",
+                "quiet",
+                "-print_format",
+                "json",
+                "-show_format",
+                "-show_streams",
+                filename,
+            ],
+            capture_output=True,
+            check=True,
+        )
+
+        data = json.loads(result.stdout)
+
+        duration = None
+        if "format" in data and "duration" in data["format"]:
+            duration = float(data["format"]["duration"])
+        else:
+            for stream in data.get("streams", []):
+                if "duration" in stream:
+                    duration = float(stream["duration"])
+                    break
+
+        return duration
+
+    @staticmethod
+    async def async_convert_mp4_to_ts(mp4_file, ts_file):
+        command = [
+            "ffmpeg",
+            "-i",
+            mp4_file,
+            "-c:v",
+            "libx264",
+            "-c:a",
+            "aac",
+            "-f",
+            "mpegts",
+            "-bsf:v",
+            "h264_mp4toannexb",
+            "-bsf:a",
+            "aac_adtstoasc",
+            ts_file,
+        ]
+
+        process = await asyncio.create_subprocess_exec(
+            *command, stdout=asyncio.subprocess.PIPE, stderr=asyncio.subprocess.PIPE
+        )
+
+        _, stderr = await process.communicate()
+
+        if process.returncode != 0:
+            error_message = stderr.decode().strip()
+            raise RuntimeError(f"FFmpeg command failed: {error_message}")
+
+        return ts_file
+
+    async def stream_output(
+        self,
+        value: str | None,
+        output_id: str,
+        first_chunk: bool,  # noqa: ARG002
+    ) -> tuple[MediaStreamChunk | None, dict]:
+        output_file = {
+            "video": {
+                "path": output_id,
+                "is_stream": True,
+                "orig_name": "video-stream.ts",
+            }
+        }
+        if value is None:
+            return None, output_file
+
+        ts_file = value
+        if not value.endswith(".ts"):
+            if not value.endswith(".mp4"):
+                raise gr.Error(
+                    "Video must be in .mp4 or .ts format to be streamed as chunks",
+                    visible=False,
+                )
+            ts_file = value.replace(".mp4", ".ts")
+            await self.async_convert_mp4_to_ts(value, ts_file)
+
+        duration = self.get_video_duration_ffprobe(ts_file)
+        if not duration:
+            raise gr.Error("Cannot determine video chunk duration", visible=False)
```

Comment:
internal gradio processes should raise gr.Error, because gr.Error bypasses `show_errors=False`. gr.Error is used in gradio apps to explicitly say they want an error displayed in the frontend

Reply:
I'm getting in the habit of using gr.Error because it's the only way the gradio clients will get an error message if something goes wrong. I thought it'd be ok since we're setting `visible=False` so it won't be shown in the UI.

In this case I can just use a normal exception since the clients don't support media streaming yet.